rasa/__init__.py

```diff
@@ -1,12 +1,148 @@
 import logging
+import typing
 
 from rasa import version
+import rasa.shared.constants as __shared_constants
+
+if typing.TYPE_CHECKING:
+    from typing import Any, Text, Dict, Union, List, Optional
+    from rasa.train import TrainingResult
+    import asyncio
 
 # define the version before the other imports since these need it
 __version__ = version.__version__
 
-from rasa.run import run
-from rasa.train import train
-from rasa.test import test
 
 logging.getLogger(__name__).addHandler(logging.NullHandler())
+
+
+def run(
```

Comment:
Should we add a comment / module docstring which explains why we pulled out this functionality? This would avoid that somebody adds stuff without considering the import behavior to this module in the future.

On another note: I'd slighly prefer to have this behavior in a named module and just import in `__init__` from there. This would make the code more discoverable and align with best practices re the `__init__.py` usage.

Reply:
yes I was thinking about this as well and you are right better to move them to a separate file

Reply:
moved them to `api.py` and added a blurp about the imports

Reply:
nice, thank you!

Reply:
doesn't work though - windows tests are failing. looks like we have name clash now (`run` module and function at `rasa.run` - thinking about it I am not even sure why this worked before...)

Reply:
Yeah it wasn't the best decision back then to name the modules the same as the most important function 🤦🏻